VideoPlayer/tests/Video-specs.js

```diff
@@ -27,237 +20,255 @@ describe('VideoPlayer.Video', () => {
-				<Video source="abc.mp4" />
+			const {rerender} = render(
+				<Video data-testid="video-id" source="abc.mp4" />
 			);
 
-			subject.setProps({
-				preloadSource: 'def.mp4'
-			});
+			rerender(
+				<Video data-testid="video-id" preloadSource="def.mp4" source="abc.mp4" />
+			);
 
-			const expected = 2;
-			const actual = subject.find('Media').length;
+			const actual = screen.getByTestId('video-id').nextElementSibling;
 
-			expect(actual).toBe(expected);
+			expect(actual).not.toBeNull();
 		});
 
 		test('should use the same node when adding `preloadSource` the same as source', () => {
-			const subject = mount(
-				<Video source="abc.mp4" />
+			const {rerender} = render(
+				<Video data-testid="video-id" source="abc.mp4" />
 			);
 
-			const expected = getSourceNode(subject);
+			const expected = screen.getByTestId('video-id');
 
-			subject.setProps({
-				preloadSource: 'abc.mp4'
-			});
+			rerender(
+				<Video data-testid="video-id" preloadSource="abc.mp4" source="abc.mp4" />
+			);
 
-			const actual = getSourceNode(subject);
+			const actual = screen.getByTestId('video-id');
 
 			expect(actual).toBe(expected);
 		});
 
 		test('should use the same node when changing `preloadSource`', () => {
-			const subject = mount(
-				<Video source="abc.mp4" preloadSource="def.mp4" />
+			const {rerender} = render(
+				<Video data-testid="video-id" preloadSource="def.mp4" source="abc.mp4" />
 			);
 
-			const expected = getPreloadNode(subject);
+			const expected = screen.getByTestId('video-id').nextElementSibling;
 
-			subject.setProps({
-				preloadSource: 'ghi.mp4'
-			});
+			rerender(
+				<Video data-testid="video-id" preloadSource="ghi.mp4" source="abc.mp4" />
+			);
 
-			const actual = getPreloadNode(subject);
+			const actual = screen.getByTestId('video-id').nextElementSibling;
 
 			expect(actual).toBe(expected);
 		});
 
 		test('should swaps nodes when swapping `source` and `preloadSource`', () => {
-			const subject = mount(
-				<Video source="abc.mp4" preloadSource="def.mp4" />
+			const {rerender} = render(
+				<Video data-testid="video-id" preloadSource="def.mp4" source="abc.mp4" />
 			);
 
-			const source = getSourceNode(subject);
-			const preload = getPreloadNode(subject);
+			const source = screen.getByTestId('video-id');
+			const preload = screen.getByTestId('video-id').nextElementSibling;
 
-			subject.setProps({
-				source: 'def.mp4',
-				preloadSource: 'abc.mp4'
-			});
+			rerender(
+				<Video data-testid="video-id" preloadSource="abc.mp4" source="def.mp4" />
+			);
 
-			expect(getSourceNode(subject)).toBe(preload);
-			expect(getPreloadNode(subject)).toBe(source);
-		});
+			const newSource = screen.getByTestId('video-id');
+			const newPreload = screen.getByTestId('video-id').nextElementSibling;
 
+			expect(newSource).toBe(preload);
+			expect(newPreload).toBe(source);
+		});
 
 		test('should not swap nodes on re-render after swapping `source` and `preloadSource`', () => {
```

Comment:
This test has been deleted. It cannot be migrated to react testing library. Re-rendering <Video /> with no props doesn't allow accessing the initial value for those props like in the enzyme test.